test/unit/lib/runner/writers/action-json-writer-tests.js

```diff
@@ -0,0 +1,170 @@
+      };
+
+      report = {
+        testReports: [
+          {
+            testRuns: [
+              {
+                report: {
+                  testName: 'testOne',
+                  actions: [
+                    {
+                      component: 'navigate',
+                      action: 'navigate',
+                    },
+                  ],
+                },
+              },
+              {
+                report: {
+                  testName: 'testOne',
+                  actions: [
+                    {
+                      component: 'modal',
+                      action: 'click',
+                    },
+                  ],
+                },
+              },
+            ],
+          },
+          {
+            testRuns: [
+              {
+                report: {
+                  testName: 'testOne',
+                  actions: [
+                    {
+                      component: 'navigate',
+                      action: 'navigate',
+                    },
+                  ],
+                },
+              },
+              {
+                report: {
+                  testName: 'testOne',
+                  actions: [
+                    {
+                      component: 'modal',
+                      action: 'click',
+                    },
+                  ],
+                },
+              },
+            ],
+          },
+        ],
+      };
+
+      fs = {
+        writeFileSync: sinon.stub(),
+        readFileSync: sinon.stub(),
+      };
+
+      configHandler = {
+        get: sinon.stub(),
+      };
+
+      obj = {};
+
+      mockery.registerMock('fs', fs);
+      mockery.registerMock('path', path);
+      mockery.registerMock('../../util/config-handler.js', configHandler);
+
+      writeActionReportToDisk = require('../../../../../lib/runner/writers/action-json-writer.js');
+    });
+
+    afterEach(function() {
+      mockery.resetCache();
+      mockery.deregisterAll();
+      mockery.disable();
+    });
+
+    it('should call path.resolve with the configs testPath', function() {
+      let report = {
+        testReports: [],
+      };
+      configHandler.get.returns('./testPath');
+
+      writeActionReportToDisk(report);
+
+      expect(path.resolve.args).to.deep.equal([
+        [
+          './testPath',
+        ],
+      ]);
+    });
+
+    describe('for each test report of the passed in report', function() {
+      it('should call fs.readFileSync with the path to the test and \'utf8\'', function() {
+        path.resolve.returns('./testpath');
+        fs.readFileSync.returns('{}');
+
+        writeActionReportToDisk(report);
+
+        expect(fs.readFileSync.args).to.deep.equal([
+          [
+            './testpath/testOne',
+            'utf8',
+          ],
+          [
+            './testpath/testOne',
+            'utf8',
+          ],
+        ]);
+      });
+
+      it('should assign customReport', function() {
+        let customReport = {
```

Comment:
this test is not doing anything, but checking that the customReport you just declared is the report you just declared. Even though the varible name has the same name as inside the function, they are seperate scopes. To check this was assigned correctly you need to check where it used, in your case on the call to when its pulled out of your 'obj' and used in side the write for fs.writeFileSync